src/main/java/com/internet/shop/dao/interfaces/ShoppingCartDao.java

```diff
@@ -0,0 +1,6 @@
+package com.internet.shop.dao.interfaces;
+
+import com.internet.shop.model.ShoppingCart;
+
+public interface ShoppingCartDao extends GenericDao<ShoppingCart, Long> {
+}
```

Comment:
is i'm not mistaken we should have here one more method: getByUserId(Long userId) which different from get(Long id)

Reply:
Added to dao and to service layer getByUserId()